tuna/machine_management_interface.py

```diff
@@ -217,37 +253,36 @@ def run_ipmi_command(self, command):
                 GATEWAY_IP, GATEWAY_PORT, GATEWAY_USER)
 
           ssh = MachineManagementInterface.gateway_session
-          _, out_ch, err_ch = ssh.exec_command(cmd, timeout=SSH_TIMEOUT)
-          error = [x.strip() for x in err_ch.readlines()]
-          error = '\n'.join(error)
-          exit_status = out_ch.channel.exit_status
+          if ssh is not None:
+            _, o_var, e_var = ssh.exec_command(cmd, timeout=SSH_TIMEOUT)
+          if e_var is not None:
+            error = [x.strip() for x in e_var.readlines()]
+            e_result = '\n'.join(error)
+            exit_status = o_var.channel.exit_status
         except paramiko.ssh_exception.SSHException:
           self.logger.warning('Failed to execute ipmitool command')
 
-      output = [x.strip() for x in out_ch.readlines()]
-      output = '\n'.join(output)
-      self.logger.warning('IPMI std output: %s', output)
-      self.logger.warning('IPMI err output: %s', error)
+      output = [x.strip() for x in o_var.readlines()]
+      rstr: str = '\n'.join(output)
+      self.logger.warning('IPMI std output: %s', rstr)
+      self.logger.warning('IPMI err output: %s', e_result)
       return exit_status
 
-  def restart_server(self):
+  def restart_server(self) -> int:
     """ Method to restart a remote machine """
-    ret = None
-    if self.backend == MgmtBackend.IPMI:
-      ret = self.run_ipmi_command("chassis power cycle")
-    else:
-      ret = self.run_bmc_command("chassisoff")
-      ret |= self.run_bmc_command("chassison")
-    return ret
+    ret_ipmi: int
+    ret_bmc: int
+    ret_status: bool = False
 
-  def server_status(self):
-    """ Return the status of the management backend of the remote machine """
-    ret = None
     if self.backend == MgmtBackend.IPMI:
-      ret = self.run_ipmi_command("chassis status")
+      ret_ipmi = self.run_ipmi_command("chassis power cycle")
+      ret_status = True
     else:
-      ret = self.run_bmc_command("chassisstate")
-    return ret
+      ret_bmc = self.run_bmc_command("chassisoff")
+      ret_bmc |= self.run_bmc_command("chassison")
+    if ret_status:
+      return ret_ipmi
+    return ret_bmc
```

Comment:
the previous code only returned one variable: ret, you seem to be returning at least ret_ipmi or ret_bmc. Please only use one variable to store the return code.

Reply:
pylint std rules, does not allow us to use single name for various function returns. I had rename

Reply:
I have tested it myself, and it does work. Please see code below:
```
 def restart_server(self) -> int:
    """ Method to restart a remote machine """
    #ret_ipmi: int
    #ret_bmc: int
    ret: int
    #ret_status: bool = False

    if self.backend == MgmtBackend.IPMI:
      ret = self.run_ipmi_command("chassis power cycle")
      #ret_status = True
    else:
      ret = self.run_bmc_command("chassisoff")
      ret |= self.run_bmc_command("chassison")
    #if ret_status:
    #  return ret_ipmi
    return ret

```

Reply:
Yes, It do. If we look at code readability persepective,  as the functions called are different having same name may confuse. To avoid this, I renamed them depending on the function call.

Reply:
@technicalgrp89 Renaming variables due to the function being called does not make sense to. Since the variable is essentially returning the status of the current function, there should be only one variable. Moreover, this reduces the lines of code considerably, therefore I think what Alex is suggesting makes sense here.

Reply:
ret value changed to Optional[int]